### ISSUE TITLE
Loss calculation in forward function of softmax class

### ISSUE THREAD

**Issue by Shehz** [chihkuanyeh/Representer_Point_Selection#3]
Hi,

From the paper & in the later part of code, I could understand that softmax cross entropy is being used as the loss function. 
`

>    def forward(self, x, y):
>         # calculate loss for the loss function and L2 regularizer
>         D = (torch.matmul(x,self.W))
>         D_max,_ = torch.max(D,dim = 1, keepdim = True)
>         D = D-D_max
>         A = torch.log(torch.sum(torch.exp(D),dim = 1))
>         B = torch.sum(D*y,dim=1)
>         Phi = torch.sum(A-B)
>         W1 = torch.squeeze(self.W)
>         L2 = torch.sum(torch.mul(W1, W1))
>         return (Phi,L2)

`
However, I couldn't relate the above implementation with the standard cross entropy loss. when we substitute the numerical stable softmax value into the cross entropy function, 
 A = y* torch.log(torch.sum(torch.exp(D),dim = 1)), in the above y term is missing. Is this a more stable version of softmax cross entropy calculation?  It will be very helpful, if you would share any information regarding the above implementation. 
Thanks  in advance!

**Comment by chihkuanyeh** [comment]
I believe our implementation is equivalent to standard cross-entropy with softmax. Is the problem resolved yet.

**Comment by Shehz** [comment]
Thank you for your comment ! Yes, the problem is solved.
